release notes/upcoming.md

```diff
@@ -54,6 +54,27 @@ Thanks to @cstyan for their work on this!
 
 **Docs**: [Request information](https://docs.k6.io/docs/TODO)
 
+### k6/http: Support for HTTP Digest Authentication (#527)
```

Comment:
The number in the brackets should link to this PR, it currently links to the issue :)